virttest/migration.py

```diff
@@ -297,7 +297,10 @@ def do_migration(self, vms, srcuri, desturi, migration_type,
                         elif func == virsh.migrate_postcopy:
                             func(vm.name, uri=srcuri, debug=True)
                         else:
-                            func(args['func_params'])
+                            try:
+                                func(args['func_params'])
+                            except (TypeError, KeyError):
```

Comment:
This is not safe because if the func(xxx) function codes raise a TypeError or KeyError, it will also be caught here.
So I still suggest we do 
1.  If  args haskey func_params, then func(args['func_params']). else func()
2.  In tp-libvirt, current cases have two scenarios: 
     a. without func_params= provided
     b. with func_params provided 
For a, no impact because there is no 'func' is provided.
For b, no impact because there is  func_params provided.
But new cases from now on can invoke like this:
```
extra_args = {}
extra_args.update({'func_params': params})  # if one case needs it
migration_test.do_migration(vms, None, dest_uri, 'orderly',
                                            options, thread_timeout=900,
                                            ignore_status=True, virsh_opt=virsh_opt,
                                            func=func_name, extra_opts=extra,
                                           **extra_args)
```

Reply:
updated